lib/snapshot_logger/snapshot_logger.ex

```diff
@@ -0,0 +1,129 @@
+defmodule SnapshotLogger.SnapshotLogger do
+  @moduledoc """
+  Logs algorithm snapshots on a cadence.
+  """
+  use GenServer
+  require Logger
+  alias Alerts.Alert
+  alias AlertsViewer.DelayAlertAlgorithm
+  alias Routes.{Route, RouteStatsPubSub}
+
+  @cadence 5 * 60 * 1000
+
+  @spec start_link(any) :: :ignore | {:error, any} | {:ok, pid}
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, opts)
+  end
+
+  @impl GenServer
+  def init(opts) do
+    delay_alert_algorithm_modules =
+      Application.get_env(:alerts_viewer, :delay_alert_algorithm_components)
+
+    bus_alerts = Alerts.subscribe() |> filtered_by_bus() |> filtered_by_delay_type()
+    bus_routes = Routes.all_bus_routes(Keyword.get(opts, :route_opts, []))
+    stats_by_route = RouteStatsPubSub.subscribe()
+    routes_with_current_alerts = Enum.filter(bus_routes, &delay_alert?(&1, bus_alerts))
+    schedule_logs()
+
+    {:ok,
+     %{
+       bus_routes: bus_routes,
+       stats_by_route: stats_by_route,
+       routes_with_current_alerts: routes_with_current_alerts,
+       delay_alert_algorithm_modules: delay_alert_algorithm_modules,
+       routes_with_recommended_alerts: []
```

Comment:
question: Do we need `routes_with_recommended_alerts` in the state? I'm not seeing anywhere we're using it, but maybe I'm missing something.

Reply:
You're right, we don't, I just forgot to take it out here.